src/structure/module.rs

```diff
@@ -59,3 +64,115 @@ pub trait VectorSpace<S: Field>
     : VectorSpaceApprox<S>
     + Module<S>
 {}
+
+
+/// A normed approximate vector space.
+pub trait NormedSpaceApprox<S: FieldApprox>
+    : VectorSpaceApprox<S> {
```

Comment:
Yes, that works but the user would be forced to add this where clause every time he uses `EuclideanSpaceApprox` in generic code. For example:

```rust
fn foo<P: EuclideanSpaceApprox>(p: &P)
where P::Scalar: RealApprox { // Obligatory otherwise the compiler complains.
}
```